src/coreclr/jit/importervectorization.cpp

```diff
@@ -648,7 +648,24 @@ GenTree* Compiler::impUtf16SpanComparison(StringComparisonKind kind, CORINFO_SIG
             JITDUMP("UTF16 data is too long to unroll - bail out.\n");
             return nullptr;
         }
-        JITDUMP("Trying to unroll MemoryExtensions.Equals|SequenceEqual|StartsWith(op1, \"%ws\")...\n", str)
+
+#ifdef DEBUG
+        if (verbose)
+        {
+            const char* utf8Str = "<utf8 conversion failure>";
+            int         utf8Len = WideCharToMultiByte(CP_UTF8, 0, (WCHAR*)str, -1, nullptr, 0, nullptr, nullptr);
+            if (utf8Len != 0)
+            {
+                char* allocated = new (this, CMK_DebugOnly) char[utf8Len];
```

Comment:
I assume this doesn't need to be freed?

Reply:
Nope, this is using the JIT arena allocator that is freed at the end of the compilation.